tensorflow/compiler/mlir/hlo/lib/Dialect/mhlo/IR/hlo_ops.cc

```diff
@@ -1405,6 +1405,58 @@ static LogicalResult Verify(ConcatenateOp op) {
   return success();
 }
 
+LogicalResult ConcatenateOp::reifyReturnTypeShapes(
+    OpBuilder& builder, ValueRange operands,
+    SmallVectorImpl<Value>& reifiedReturnShapes) {
+  ConcatenateOp::Adaptor adaptor(operands);
+  auto inputs = adaptor.val();
+  size_t axis = static_cast<size_t>(
+      this->getOperation()->getAttrOfType<IntegerAttr>("dimension").getInt());
```

Comment:
You should be able to use `this->dimension()` here?
Also I don't think the static_cast is needed in the assignment, is it?
Since `axis` is only used as subscript, you can likely use `int` alone.

Finally, please move all variable assignment closer to their uses, so this code isn't executed on the early exit paths (like the return 4 lines below).

Reply:
Thanks! I have fixed it.